### PR TITLE
Implemented uWSGI formula

### PR DESCRIPTION
**Pull request by blarghmatey**
Added states to:
- Install uWSGI, allowing for a flexible build to use different profiles
- Run an emperor process via SystemD
- Configure one or more application configurations via pillar data
- Add additional configuration to the emperor process

### CODE REVIEW

File: pillar.example [hunk 1]
```diff
@@ -1,6 +1,14 @@
+# -*- mode: yaml -*-
 uwsgi:
-  lookup:
-    pkg: myuwsgi
-    conf_file: /usr/local/etc/uwsgi/uwsgi.conf
-  config:
-    key: othervalue
+  emperor_config:
```
Comment (shaidar) [inline pillar.example:3]: Doesn't this need to be under an `overrides` section?

File: uwsgi/files/emperor.ini [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[uwsgi]
+emperor = /etc/uwsgi/vassals/
+if-exists = /etc/uwsgi/emperor_extra.ini
+include = %(_)
+endif =
```
Comment (shaidar) [inline uwsgi/files/emperor.ini:5]: New line

File: uwsgi/files/uwsgi.service [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# Requires systemd version 211 or newer
+RuntimeDirectory=uwsgi
+Restart=always
+KillSignal=SIGQUIT
+Type=notify
+NotifyAccess=all
+ExecReload=/usr/local/bin/uwsgi --reload
+
+[Install]
+WantedBy=multi-user.target
```
Comment (shaidar) [inline uwsgi/files/uwsgi.service:16]: New line

**Comment by shaidar** [comment]
Was just walking through the formula and noticed [this](https://github.com/mitodl/uwsgi-formula/blob/tmacey/uwsgi_install/uwsgi/files/emperor.ini#L5). Is this okay without a value?

File: pillar.example (hunk 1 above)
Comment (blarghmatey) [reply pillar.example:3]: No, because it is referenced directly in the config.sls state file.

**Comment by blarghmatey** [comment]
Yes, that is the expected syntax for conditional blocks in the INI file. http://uwsgi-docs.readthedocs.io/en/latest/ConfigLogic.html

**Comment by blarghmatey** [comment]
That is also how you can set parameters that don't require any input.

**Comment by shaidar** [comment]
:thumbsup:

**Pull request merged**